src/org/opensolaris/opengrok/search/context/QueryMatchers.java

```diff
@@ -89,9 +90,19 @@ private void getTerms(Query query) {
             getTerm((TermQuery) query);
         } else if (query instanceof PrefixQuery) {
             getPrefix((PrefixQuery) query);
+        } else if (query instanceof RegexpQuery) {
```

Comment:
confirming, you need to patch Summarizer too - it creates the short descriptions in the listing of found docs
(or long ones? )
anyways, it needs to be there, but it will be similar to what you did now
(I think Summarizer was the reason why I didn't fix this bug 2 years ago ... I run out of time on it and then didn't return to that patch since :( )